### PR TITLE
add bounds to style-spec docs

### LINKED ISSUE #5124: Add Source#bounds to style spec
https://github.com/mapbox/mapbox-gl-js/pull/4556 added a `bounds` property to the source description, similar to what [TileJSON supports](https://github.com/mapbox/tilejson-spec/tree/master/2.2.0). We should add this property to the style spec explicitly.

/cc @mollymerp

### PR DESCRIPTION
**Pull request by mollymerp**
close #5124 

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page

### CODE REVIEW

File: src/style-spec/reference/v8.json [hunk 1]
```diff
@@ -135,6 +135,11 @@
       "value": "string",
       "doc": "An array of one or more tile source URLs, as in the TileJSON spec."
     },
+    "bounds":{
+      "type":"array",
```
Comment (jfirebaugh) [inline src/style-spec/reference/v8.json]: Nit: space after `:` (x2).

File: src/style-spec/reference/v8.json [hunk 2]
```diff
@@ -135,6 +135,11 @@
...
+      "value":"number",
```
Comment (jfirebaugh) [inline src/style-spec/reference/v8.json]: Also include `"length": 4`.

File: src/style-spec/reference/v8.json [hunk 3]
```diff
@@ -135,6 +135,13 @@
       "value": "string",
       "doc": "An array of one or more tile source URLs, as in the TileJSON spec."
     },
+    "bounds":{
```
Comment (kkaefer) [inline src/style-spec/reference/v8.json]: Nit: space before `{`

File: src/source/vector_tile_source.js [hunk 4]
```diff
@@ -73,8 +74,8 @@ class VectorTileSource extends Evented implements Source {
     }
 
     setBounds(bounds?: [number, number, number, number]) {
-        this.bounds = bounds;
         if (bounds) {
```
Comment (jfirebaugh) [inline src/source/vector_tile_source.js]: I think the intent might have been to allow `setBounds(null)` or `setBounds(undefined)` to remove an existing bounds. It doesn't look like that's actually used anywhere, however, and `setBounds` isn't a public API. I'd say we should just inline `setBounds` where it's used in the `loadTileJSON` in `load`, and set the bounds in the initial source properties where they're needed in `{raster,vector}_tile_source.test.js`.

Relatedly, I don't think it should be necessary to add any calls to `setBounds`, as you've done above. `loadTileJSON` calls the callback in both cases: when the initial source properties contain inline TileJSON properties, and when they contain a `url` to TileJSON.

File: test/integration/render-tests/regressions/mapbox-gl-js#4564/style.json [hunk 5]
```diff
@@ -18,7 +18,7 @@
       "tiles": [
         "local://tiles/{z}-{x}-{y}.mvt"
       ],
-      "bounds":[[13.4043671, 52.4959407], [13.43183291,52.502459]]
+      "bounds":[13.4043671, 52.4959407, 13.43183291,52.502459]
```
Comment (jfirebaugh) [inline test/integration/render-tests/regressions/mapbox-gl-js#4564/style.json]: Yay validation!

File: src/source/vector_tile_source.js (hunk 4 above)
Comment (mollymerp) [inline src/source/vector_tile_source.js]: 👍  much simpler!

**Pull request merged**